src/topic.ts

```diff
@@ -326,23 +360,33 @@ export class Topic {
    *   const apiResponse = data[1];
    * });
    */
-  get(gaxOpts, callback) {
-    if (is.fn(gaxOpts)) {
-      callback = gaxOpts;
-      gaxOpts = {};
-    }
+  get(callback: CreateTopicCallback): void;
+  get(gaxOpts?: CallOptions&GetCallOptions): Promise<Topic>;
+  get(gaxOpts: CallOptions&GetCallOptions, callback: CreateTopicCallback): void;
+  get(gaxOptsOrCallback?: CallOptions&GetCallOptions|CreateTopicCallback,
+      callback?: CreateTopicCallback): void|Promise<Topic> {
+    const gaxOpts =
+        typeof gaxOptsOrCallback === 'object' ? gaxOptsOrCallback : {};
+    callback =
+        typeof gaxOptsOrCallback === 'function' ? gaxOptsOrCallback : callback;
     const autoCreate = !!gaxOpts.autoCreate;
     delete gaxOpts.autoCreate;
     this.getMetadata(gaxOpts, (err, apiResponse) => {
       if (!err) {
-        callback(null, this, apiResponse);
+        callback!(null, this, apiResponse!);
         return;
       }
-      if (err.code !== 5 || !autoCreate) {
-        callback(err, null, apiResponse);
+      let code = 0;
+      if (err.hasOwnProperty('code')) {
```

Comment:
same, lets drop the weird rewrite if we can

Reply:
Done!